cloudfront/wellcomelibrary.org/edge-lambda/src/api.ts

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+
+const apiBasePath = 'https://api.wellcomecollection.org/catalogue/v2';
+
+export type ApiQuery = {
+  query: string;
+  include: string[];
+};
+
+export type IdentifierType = {
+  id: string;
+  label: string;
+  type: 'IdentifierType';
+};
+
+export type Identifier = {
+  value: string;
+  identifierType: IdentifierType;
+  type: 'Identifier';
+};
+
+export type Work = {
+  type: 'Work' | 'Collection' | 'Section' | 'Series';
+  id: string;
+  title: string;
+  identifiers?: Identifier[];
+};
+
+export type CatalogueResultsList<Result = Work> = {
+  type: 'ResultList';
+  totalResults: number;
+  results: Result[];
+  pageSize: number;
+  prevPage?: string;
+  nextPage: string | undefined;
+};
+
+export type GetWorkResult = Work | Error;
+
+export async function* apiQuery(
+  query: ApiQuery
+): AsyncGenerator<Work, void, void> {
```

Comment:
Could you explain a little about why you chose to use an async generator here? I've only come across them very rarely in application code (although that doesn't mean it's a bad choice) with the exception of some libraries that lean on them heavily.

Reply:
It's not really a carefully considered choice. Thinking "I need something like a generator here, and the underlying requests should be async", cue me googling "javascript async generator".